### ISSUE TITLE
async-start generates a "*emacs*:err" buffer

### ISSUE THREAD

**Issue by xiliuya** [jwiegley/emacs-async#173]
I tested under `emacs - Q`
When I run the following code in `emacs 29` an error buffer is generated.

```
(async-start
       ;; What to do in the child process
       (lambda ()
         (message "This is a test")
         (sleep-for 3)
         222)

       ;; What to do when it finishes
       (lambda (result)
         (message "Async process done, result should be 222: %s"
                  result)))

```

The buffer contents are as follows:

```
This is a test

Process emacs stderr finished
```

Is this normal?

How can I avoid generating such a buffer?
------
The version I use is as follows:
OS:  Archlinux
version:  GNU Emacs 29.0.60 (build 25, x86_64-pc-linux-gnu, GTK+ Version 3.24.37, cairo version 1.17.8) of 2023-03-19)
async version: async-20230314.1202

**Comment by thierryvolpiatto** [comment]
> Is this normal?
I guess yes, see PR #167 , perhaps @Fuco1 can give you more detailed explanations.
> How can I avoid generating such a buffer?
Is this really annoying?

**Comment by xiliuya** [comment]
>  The error buffer should automatically disappear after the subprocess finishes, is that not the case? 

@Fuco1  No, this buffer will always exist after the process ends.

**Comment by xiliuya** [comment]
> > Is this normal?
> > I guess yes, see PR #167 , perhaps @Fuco1 can give you more detailed explanations.
> > How can I avoid generating such a buffer?
> > Is this really annoying?

@thierryvolpiatto Yes, when I execute such code 10 times, there will be 10 such buffers.
As shown in the following figure: 
![2023-03-21-095315](https://user-images.githubusercontent.com/67564860/226501792-26a76042-9c97-464c-b38b-a1ccb3b8ed3d.png)

**Comment by xiliuya** [comment]
I modified `async.el` to have only one `*emacs*: error` buffer. This solved my problem. 
Thank you for your help!
@Fuco1 @thierryvolpiatto 
```diff
404c404
<          (buf-err (generate-new-buffer (concat "*" name "*:err")))
---
>          (buf-err (get-buffer-create (concat "*" name "*:err")))

```

**Comment by xiliuya** [comment]
Another way is to detect async-debug:

```diff
404c404,405
<          (buf-err (generate-new-buffer (concat "*" name "*:err")))
---
>          (buf-err (unless (null async-debug)
>                     (generate-new-buffer (concat "*" name "*:err"))))

```

**Comment by thierryvolpiatto** [comment]
xiliuya ***@***.***> writes:

> 1.  ( ) text/plain          (*) text/html           
>
>         Is this normal?
>         I guess yes, see PR #167 , perhaps @Fuco1 can give you more detailed explanations.
>         How can I avoid generating such a buffer?
>         Is this really annoying?
>        
> @thierryvolpiatto Yes, when I execute such code 10 times, there will
> be 10 such buffers.

Yes but it is a testing code and it runs a message in the child Emacs,
there is no need to call message unless you want to debug something (it
will not be seen in parent emacs), you should not have such message when
e.g. copying a file async from Helm or Dired.
So having such new buffer at each async session when a message is sent
is not a bad thing IMHO, it allows to debug your code much easily as before. 

> As shown in the following figure:
> 2023-03-
>
> —
> Reply to this email directly, view it on GitHub, or unsubscribe.
> You are receiving this because you were mentioned.*Message ID: ***@***.***>


-- 
Thierry

**Comment by thierryvolpiatto** [comment]
Matus Goljer ***@***.***> writes:

> If so, we should take care to get rid of it except maybe when
> async-debug is on.

Done.

Thanks.


-- 
Thierry

**Comment by Fuco1** [comment]
@thierryvolpiatto This is not good, the buffer must always exist to separate the output streams.  Rather we should make sure to kill it afterwards.  There seems to be similar code to kill the regular process buffer. I guess emacs keeps them around for you to "inspect" the command output.

**Comment by pataquets** [comment]
Confirming as fixed for 28.1. Thanks!

**Comment by Fuco1** [comment]
It's not fixed, it breaks other features.  By the way, the buffers automatically close for me on Emacs 28.2.  Do you have some special setting to keep them around?  Even on `emacs -Q` on 28 they disappear automatically.
